consensus/service/src/byzantine_ledger/mod.rs

```diff
@@ -63,146 +86,113 @@ impl ByzantineLedger {
         broadcaster: Arc<Mutex<dyn Broadcast>>,
         msg_signer_key: Arc<Ed25519Pair>,
         tx_source_urls: Vec<String>,
-        opt_scp_debug_dump_dir: Option<PathBuf>,
+        scp_debug_dir: Option<PathBuf>,
         logger: Logger,
     ) -> Self {
-        let (sender, receiver) =
-            mc_util_metered_channel::unbounded(&counters::BYZANTINE_LEDGER_MESSAGE_QUEUE_SIZE);
-
-        let current_slot_index = ledger.num_blocks().unwrap();
-
-        let scp_node = {
+        // TODO: this should be passed in as an argument.
+        let scp_node: Box<dyn ScpNode<TxHash>> = {
             let tx_manager_validate = tx_manager.clone();
             let tx_manager_combine = tx_manager.clone();
-            Node::new(
+            let current_slot_index = ledger.num_blocks().unwrap();
+            let node = Node::new(
                 node_id.clone(),
-                quorum_set.clone(),
+                quorum_set,
                 Arc::new(move |tx_hash| tx_manager_validate.validate(tx_hash)),
                 Arc::new(move |tx_hashes| tx_manager_combine.combine(tx_hashes)),
                 current_slot_index,
                 logger.clone(),
-            )
-        };
+            );
 
-        let wrapped_scp_node: Box<dyn ScpNode<TxHash>> = if let Some(path) = opt_scp_debug_dump_dir
-        {
-            Box::new(
-                LoggingScpNode::new(scp_node, path, logger.clone())
-                    .expect("Failed creating LoggingScpNode"),
-            )
-        } else {
-            Box::new(scp_node)
+            match scp_debug_dir {
+                None => Box::new(node),
+                Some(path) => Box::new(
+                    LoggingScpNode::new(node, path, logger.clone())
+                        .expect("Failed creating LoggingScpNode"),
+                ),
+            }
         };
 
-        let highest_outgoing_consensus_msg = Arc::new(Mutex::new(None));
+        // The worker's task queue.
+        // TODO: this should be a bounded channel.
```

Comment:
Why? What do we get from that? How would you set the limit?

Reply:
I'm mostly thinking about "failing gracefully" when something goes wrong. For example, something has probably goes wrong if the channel contains, say, 1M or 10M messages. In that case, it would be better for channel.try_send to fail (and, presumably, we'd log an error message) than for the channel to continue growing and using up memory.